apps/docs/pages/guides/database/query-optimization.mdx

```diff
@@ -0,0 +1,134 @@
+- Note that if `a.id` is the primary key of the `customers` table it is already indexed
+
+```sql
+create index idx_orders_customer_id on orders (customer_id);
+```
+
+### `order by` Clause
+
+Sorting can also be optimized by indexing:
+
+- An index on `b.date_of_purchase` wcan improve the sorting process, and is particularly beneficial when a subset of rows is being returned with a `limit` clause.
+
+```sql
+create index idx_orders_date_of_purchase on orders (date_of_purchase);
+```
+
+## Key Concepts
+
+Here are some concepts and tools to keep in mind to help you identify the best index for the job, and measure the impact that your index had:
+
+### Analyze the Query Plan
+
+Use the `explain` command to understand the query's execution. Look for slow parts, such as Sequential Scans or high cost numbers. If creating an index does not reduce the cost of the query plan, remove it.
+
+For example:
+
+```sql
+explain select * from customers where sign_up_date > 25;
+```
+
+### Use Appropriate Index Types
+
+PostgreSQL offers various index types like [B-tree, Hash, GIN, etc](https://www.postgresql.org/docs/current/indexes-types.html). Select the type that best suits your data and query pattern. Using the right index type can make a significant difference. For example, using a BRIN index on field that is always increasing on a table with infrequent updates, like `created_at` on an `orders` table, routinely results in indexes that are +10x smaller than the equivalent default B-tree index. That translates into better scalability.
+
+```sql
+create index idx_orders_created_at ON customers using brin(created_at);
+```
+
+### Partial Indexes
+
+For queries that frequently target a subset of data, a partial index could be faster and smaller than indexing the entire column. A partial index contains a `where` clause to filter the values included in the index. Note that a query's `where` clause must match the index for it to be used.
+
+```sql
+create index idx_orders_status on orders (status)
+where status = 'shipped';
+```
+
+### Composite Indexes
+
+If filtering or joining on multiple columns, a composite index can help postgres referring to multiple indexes to identify the relevant rows.
+
+```sql
+create index idx_customers_sign_up_date_priority on customers (sign_up_date, priority);
+```
+
+### Over-Indexing
+
+Avoid the urge to index columns you operate on infrequently. While indexes can speed up reads, they also slow down writes, so its important to balance those factors when making indexing decisions.
+
+### Statistics
+
+Postgres maintains a set of statistics about the contents of your tables. Those statistics are used by the query planner to decide e.g. when its is more efficient to use an index vs scanning the entire table. If the collected statistics drift too far from reality, the query planner may make poor decisions. To avoid the risk, you can periodically `analyze` tables.
```

Comment:
```suggestion
Postgres maintains a set of statistics about the contents of your tables. Those statistics are used by the query planner to decide when it's is more efficient to use an index vs scanning the entire table. If the collected statistics drift too far from reality, the query planner may make poor decisions. To avoid this risk, you can periodically `analyze` tables.
```